File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int left = 0;
        int right = 1;
        int lastmax = 0;
        
        while(right < prices.length){
            
            if(prices[right] > prices[left]){
                
                lastmax = Math.max(prices[right] - prices[left], lastmax);
                
            }else{
                
                left = right;
                
            }
            
            right++;
            
        }
        return lastmax;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        left = 0
        right = 1
        
        maxprofit = 0
        
        
        while right < len(prices):
            
            if prices[right] > prices[left]:
                curr_profit = prices[right] - prices[left]
                maxprofit = max(maxprofit,curr_profit)
            else:
                left = right
        
            right += 1
            
            
        return maxprofit
        